### PR TITLE
Update extension docs for logs integration

### PR DESCRIPTION
**Pull request by pinkerton**
### What does this PR do?

Add docs for the Lambda Extension logs launch: https://aws.amazon.com/blogs/compute/using-aws-lambda-extensions-to-send-logs-to-custom-destinations/

### Motivation

New functionality

### Preview

https://docs-staging.datadoghq.com/stephen.pinkerton/extension-logs-docs/serverless/datadog_lambda_library/extension#log-collection

### CODE REVIEW

**Review by remeh — APPROVED** [review]

File: content/en/serverless/datadog_lambda_library/extension.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ further_reading:
 
 AWS Lambda Extensions are companion processes that augment your Lambda functions. They run within the Lambda execution environment, alongside your Lambda function code. The Datadog Extension is a lightweight version of the Datadog agent built to run alongside your code with minimal performance overhead.
 
-The Datadog Extension currently supports submitting custom metrics [synchronously][1] while your AWS Lambda function executes. This means that you can submit your custom metrics without the [Datadog Forwarder][2]. Note that the Datadog Forwarder is still required to submit logs and traces to Datadog.
+The Datadog Extension currently supports submitting custom metrics and logs [synchronously][1] while your AWS Lambda function executes. This means that you can submit some of your telemetry data without the [Datadog Forwarder][2]. Note that the Datadog Forwarder is still required to submit traces to Datadog.
```
Comment (ruthnaebeck) [inline content/en/serverless/datadog_lambda_library/extension.md:16]:
```suggestion
The Datadog Extension supports submitting custom metrics and logs [synchronously][1] while your AWS Lambda function executes. This means that you can submit some of your telemetry data without the [Datadog Forwarder][2]. **Note**: The Datadog Forwarder is still required to submit traces to Datadog.
```

File: content/en/serverless/datadog_lambda_library/extension.md [hunk 2]
```diff
@@ -39,6 +39,10 @@ The Datadog Extension is distributed as its own Lambda Layer (separate from the
 
 4. Reference the [sample code][10] to submit a custom metric.
 
+### Log Collection
```
Comment (ruthnaebeck) [inline content/en/serverless/datadog_lambda_library/extension.md:42]:
```suggestion
### Log collection
```

File: content/en/serverless/datadog_lambda_library/extension.md [hunk 3]
```diff
@@ -39,6 +39,10 @@ The Datadog Extension is distributed as its own Lambda Layer (separate from the
...
+
+To submit your AWS Lambda logs to Datadog using the Extension, set the env variable `DD_LOGS_ENABLED` to `true` on your function. This additionally generates Datadog's [enhanced metrics][11] automatically.
```
Comment (ruthnaebeck) [inline content/en/serverless/datadog_lambda_library/extension.md:44]:
```suggestion
To submit your AWS Lambda logs to Datadog using the Extension, set the env variable `DD_LOGS_ENABLED` to `true` in your function. Additionally, this generates Datadog's [enhanced metrics][11].
```

**Review by ruthnaebeck — CHANGES_REQUESTED** [review]
Docs review

**Review by ruthnaebeck — APPROVED** [review]
👍

**Pull request merged**